lang/java/reef-io/src/main/java/org/apache/reef/io/storage/FramingOutputStream.java

```diff
@@ -68,13 +68,13 @@ public void write(int b) throws IOException {
   @Override
   public void write(byte[] b) throws IOException {
     baos.write(b);
-    offset += b.length;
+    this.offset += b.length;
   }
 
   @Override
-  public void write(byte[] b, int offset, int length) throws IOException {
-    baos.write(b, offset, length);
-    offset += length;
+  public void write(byte[] b, int offsetToWrite, int length) throws IOException {
+    baos.write(b, offsetToWrite, length);
+    this.offset += length;
```

Comment:
This is behavior change, I believe this was a bug (using {{offset}} parameter instead of field)